pkg/apis/sessions/session_state.go

```diff
@@ -27,6 +27,7 @@ type SessionState struct {
 	Email             string   `msgpack:"e,omitempty"`
 	User              string   `msgpack:"u,omitempty"`
 	Groups            []string `msgpack:"g,omitempty"`
+	Roles             []string `msgpack:"r,omitempty"`
```

Comment:
Nothing new should be added to the session. You should add the roles to the existing `session.Groups` option to ride default `Authorize` & header injection implementations. This helps keep the codebase more uniform so providers can automatically get new features as we adopt them globally if they adhere to conventions.

When providers have multiple group-like structures that might want to be in this field, we've adopted prefixing the names.  Here's a sample from a very similar GitLab PR that just merged: https://github.com/oauth2-proxy/oauth2-proxy/pull/630/files#diff-9179d3c8d617839cfc99bbcd5a63a315b6cf3e652bb7f7fb9dd4bd8c6b40128aR310

Reply:
Got it, I will make the required changes.